_posts/2017-09-05-Rust-2017-Survey-Results.md

```diff
@@ -0,0 +1,126 @@
+*(hover for more info)*
+
+This year, we're seeing a growing amount of experienced users sticking with Rust, with the "more than a year" users growing to over 42% (up from 30% from last year).  The beginners are also an impressively large set, with the "less than a month" crowd at just about 18%, meaning we're currently attracting nearly a 1/5th of our user base size, even as it grows larger, every month.
+
+![Chart: 36.5% less 1000 lines, 46.3% 1000 to 10000 lines, 14.2% 10000 to 100000 lines, 2.1% over 100000, 0.9% don't know](/images/2017-09-05-Rust-2017-Survey-Results/size_of_rust_projects.png "36.5% less 1000 lines, 46.3% 1000 to 10000 lines, 14.2% 10000 to 100000 lines, 2.1% over 100000, 0.9% don't know")
+*(hover for more info)*
+
+People are working ever-larger amounts of Rust, with medium- and large-scale lines of code totals both nearly doubling since last year as a percentage of the whole, now making up 16% of respondents (up from last year's 8.9%).  This shows a growing interest in using Rust in ever-larger projects, and a growing need for tools to support this growth.
+
+![Chart: 17.5% daily, 43.3% weekly, 24.4% monthly, 14.9% rarely](/images/2017-09-05-Rust-2017-Survey-Results/how_often_use_rust.png "17.5% daily, 43.3% weekly, 24.4% monthly, 14.9% rarely")
+
+Despite the rising amount of code developers are working with, we're seeing a small downtick in both daily and weekly users.  Daily users have fallen from 19% to 17.5%, and weekly users have fallen from 48.8% to 43.3%. This could be a natural transition in this stage of our growth, as a broader range of developers begin using Rust.
+
+# Path to Stability
+
+![Chart: 92.5% no, 7.5% yes](/images/2017-09-05-Rust-2017-Survey-Results/upgrade_stable.png "92.5% no, 7.5% yes")
+
+In the last year, we made big strides in breakages caused by releases of the compiler.  Last year, 16.2% respondents said that upgrading to a new stable Rust compiler broke their code.  This year, that number has **fallen to 7.5% of respondents**.  This is a huge improvement, and one we're proud of, though we'll continue working to push this down even further.
+
+![Chart show strong support for nightly and current stable releases](/images/2017-09-05-Rust-2017-Survey-Results/which_version.png)
+
+Developers have largely opted to move to nightly or a recent stable (with some on beta), showing that developers are eager to upgrade and do so quickly.  This simplifies the support structure a bit from last year, where developers were on a wider range of versions.
+Unfortunately, despite our efforts with procedural macros and helping move crates like Serde to stable, we still have work to do to promote people moving away from the nightly Rust compiler.  This year shows an increase in nightly users, now at 1,852 votes it represents 51.6% of respondents using nightly, up from 48.8% of last year.
```

Comment:
This seems not necessarily negative? It's only bad if they don't want to be on nightly, it seems possible that people's trade off is moving towards nightly because it is more stable than last year? Or there are more enticing features.

Reply:
In general, requiring people to use nightly is seen as a negative by many folks and a sign of Rust not having the level of maturity that some corporate/conservative users look for.  A lot of the push this year for getting things like serde on stable was to help address this.